recipes/brpc/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("incubator-brpc-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        # rename brpc's FindGFLAGS.cmake to avoid picking up Findgflags.cmake
+        # generated by the cmake_find_package generator
+        os.rename(
+                os.path.join(self._source_subfolder, "cmake", "FindGFLAGS.cmake"),
+                os.path.join(self._source_subfolder, "cmake", "FindBRPCGFLAGS.cmake"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_GLOG"] = self.options.with_glog
+        self._cmake.definitions["WITH_THRIFT"] = self.options.with_thrift
+        if self.options.with_snappy:
+            self._cmake.definitions["WITH_SNAPPY"] = True
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        if not self.options['protobuf'].with_zlib:
+            raise ConanInvalidConfiguration("brpc requires protobuf:with_zlib=True")
+        if self.options.with_snappy and not self.options["leveldb"].with_snappy:
+            raise ConanInvalidConfiguration("brpc requires leveldb:with_snappy=True")
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _remove_files(self, path, patterns):
+        for file_pattern in patterns:
+            for file in glob.glob(os.path.join(path, file_pattern)):
+                os.remove(file)
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        lib_path = os.path.join(self.package_folder, "lib")
+        if self.options.shared:
+            self._remove_files(lib_path, ["*.a", "*.lib"])
+        else:
+            self._remove_files(lib_path, ["*.so", "*.dylib", "*.dll"])
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["brpc"]
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["dl", "pthread", "rt"]
+        if self.settings.os == "Macos":
+            self.cpp_info.frameworks.append("CoreFoundation")
+            self.cpp_info.frameworks.append("CoreGraphics")
+            self.cpp_info.frameworks.append("CoreText")
+            self.cpp_info.frameworks.append("Foundation")
+            self.cpp_info.frameworks.append("Security")
+            self.cpp_info.exelinkflags.append("-Wl,-U,_MallocExtension_ReleaseFreeMemory")
+            self.cpp_info.exelinkflags.append("-Wl,-U,_ProfilerStart")
+            self.cpp_info.exelinkflags.append("-Wl,-U,_ProfilerStop")
```

Comment:
Are you sure you are building with the profiler disabled?

Reply:
For the MacOS updates, I basically referred to https://github.com/apache/incubator-brpc/pull/223 for hints.

IIUC in the BRPC code, this is basically specified as a weak symbol so that there is no hard dependency on gperftools and its includes. Instead, they directly declare these function prototypes and mark them as weak and instruct the linker to allow the non-existence of its definition. So there is no linker error. The user application that uses brpc can choose to link with `-lprofiler` or use `LD_PRELOAD` to "enable" the brpc profiler service. If the profiler has not been linked in, then the underlying function pointer would remain null. The brpc profiler service [checks for this condition](https://github.com/apache/incubator-brpc/blob/master/src/brpc/builtin/pprof_service.cpp#L103) and returns a runtime error in this scenario. That said, I haven't actually used this technique or the in-built profiler service but this is my understanding on how it is set up.